File: in_Java/0119 Pascal's Triangle II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<List<Integer>> L = new ArrayList<>();
        List<Integer> res = new ArrayList<>();
        //if(rowIndex == 0) return res;
        res.add(1);
        L.add(res);
        if(rowIndex == 0) return res;  
        res = new ArrayList<>();
        res.add(1);
        res.add(1);
        L.add(res);
        List<Integer> pre;
        List<Integer> cur;
        for(int i = 2; i <= rowIndex; i++){
            pre = L.get(i-1);
            cur = new ArrayList<>();
            cur.add(1);
            for(int j = 1; j < i; j++){
                cur.add(pre.get(j)+pre.get(j-1));
            }
            cur.add(1);
            L.add(cur);
        }
        return L.get(L.size()-1);
    }
}

File: in_Python/0119 Pascal's Triangle II.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        n = rowIndex
        if n == 0:
            return [1]
        res1 = [1]
        res2 = [1]
        if n == 1:
            return res1 + res2
        for i in range(1, n/2):
            if i == 1:
                res1.append(n)
                res2.insert(0,n)
            else:
                prev = res1[-1]
                cur = prev * (n - i + 1) / i
                res1.append(cur)
                res2.insert(0,cur)
        prev = res1[-1]
        cur = prev * (n - n/2 + 1) / (n/2)
        if n % 2 == 1:
            res2.insert(0,cur)
        res1.append(cur)
        return res1 + res2